### ISSUE TITLE
Sync-usage-file results in instance costs to show 0

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#987]
This is an issue that was introduced in master, but was not in Infracost v0.9.8.

When using infracost with out internal infra that uses EKS, we have something like this:
```
module "eks" {
  source          = "terraform-aws-modules/eks/aws"
  cluster_name    = var.env_name
  cluster_version = "1.19"
  subnets         = module.vpc.private_subnets
  vpc_id          = module.vpc.vpc_id
  enable_irsa     = true

  node_groups = {
    default = {
      desired_capacity = 2
      max_capacity     = 10
      min_capacity     = 1

      instance_types = [var.cluster_instance_type]
    }
  }
}
```

This shows the instance costs ok without sync-usage-file:
```
 module.base.module.eks.module.node_groups.aws_eks_node_group.workers["default"]
 ├─ Instance usage (Linux/UNIX, on-demand, t3.medium)                                         2,920  hours                        $121.47
 ├─ CPU credits                                                                                   0  vCPU-hours                     $0.00
 └─ Storage (general purpose SSD, gp2)                                                          400  GB                            $40.00
```

In the master branch, the CLI shows the costs as 0 when used with sync-usage file. In Infracost v0.9.8, it shows a lower cost as default RI options are 1_year, standard, partial_upfront. 
```
 module.base.module.eks.module.node_groups.aws_eks_node_group.workers["default"]
 ├─ Instance usage (Linux/UNIX, on-demand, t3.medium)                                          0  hours                    $0.00
 ├─ CPU credits                                                                                0  vCPU-hours               $0.00
 └─ Storage (general purpose SSD, gp2)                                                         0  GB                       $0.00 

time="2021-09-29T10:49:53+01:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2021-09-29T10:49:53+01:00" level=warning msg="Invalid reserved_instance_type, ignoring reserved options. Expected: convertible, standard. Got: "
time="2021-09-29T10:49:53+01:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2021-09-29T10:49:53+01:00" level=warning msg="Invalid reserved_instance_type, ignoring reserved options. Expected: convertible, standard. Got: "
time="2021-09-29T10:49:53+01:00" level=info msg="starting: Syncing usage data from cloud"
time="2021-09-29T10:49:57+01:00" level=debug msg="Loading usage data from usage file"
time="2021-09-29T10:49:57+01:00" level=debug msg="Checking if . is a git repo"
time="2021-09-29T10:49:57+01:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2021-09-29T10:49:57+01:00" level=warning msg="Invalid reserved_instance_type, ignoring reserved options. Expected: convertible, standard. Got: "
time="2021-09-29T10:49:57+01:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2021-09-29T10:49:57+01:00" level=warning msg="Invalid reserved_instance_type, ignoring reserved options. Expected: convertible, standard. Got: "
```

**Comment by xeger** [comment]
With no usage defined, the terraform package derives instance count like so:
```
	instanceCount := d.Get("scaling_config.0.desired_size").Int()
```

However, when a usage file is present, the aws package populates usage before returning the built resource:
```
	a.PopulateUsage(u)
```

This overwrites the `InstanceCount` field with a 0 value, causing the cost to drop to zero.

Either we should comment out the 0 (defaulted) values we write to the YML, or when syncing usage, we should use `desired_count` as the default value.

**Comment by tim775** [comment]
Fixed by #1021
